crypto/schnorr-adaptor-signature/types.go

```diff
@@ -0,0 +1,173 @@
+package schnorr_adaptor_signature
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const (
+	ModNScalarSize       = 32
+	FieldValSize         = 32
+	JacobianPointSize    = 33
+	AdaptorSignatureSize = JacobianPointSize + ModNScalarSize
+)
+
+type AdaptorSignature struct {
+	R0 btcec.JacobianPoint
+	s  btcec.ModNScalar
+}
+
+// newAdaptorSignature creates a new AdaptorSignature with the given parameters.
+// It copies the values to avoid unexpected modifications.
+func newAdaptorSignature(r *btcec.JacobianPoint, s *btcec.ModNScalar) *AdaptorSignature {
+	var sig AdaptorSignature
+	sig.R0.Set(r)
+	sig.s.Set(s)
+	return &sig
+}
+
+// NewAdaptorSignatureFromBytes parses the given byte array to an adaptor signature.
+// The format is:
+// - First byte: 0x02 if R0.Y is even, 0x03 if R0.Y is odd
+// - Next 32 bytes: R0.X
+// - Last 32 bytes: s scalar
+func NewAdaptorSignatureFromBytes(asigBytes []byte) (*AdaptorSignature, error) {
+	if len(asigBytes) != AdaptorSignatureSize {
+		return nil, fmt.Errorf(
+			"the length of the given bytes for adaptor signature is incorrect (expected: %d, actual: %d)",
+			AdaptorSignatureSize,
+			len(asigBytes),
+		)
+	}
+
+	// Check the first byte is valid
+	if asigBytes[0] != 0x02 && asigBytes[0] != 0x03 {
+		return nil, fmt.Errorf("invalid format byte: %x", asigBytes[0])
+	}
+
+	// Extract R0.X from bytes 1-32
+	var xVal btcec.FieldVal
+	if overflow := xVal.SetByteSlice(asigBytes[1:33]); overflow {
+		return nil, fmt.Errorf("R0.X value is invalid")
+	}
+
+	// Decompress the point with the correct y-coordinate parity
+	var yVal btcec.FieldVal
+	isOdd := asigBytes[0] == 0x03
+	if success := btcec.DecompressY(&xVal, isOdd, &yVal); !success {
+		return nil, fmt.Errorf("failed to decompress R0.Y")
+	}
+
+	// Create the Jacobian point
+	var zVal btcec.FieldVal
+	zVal.SetInt(1)
+	r0 := btcec.MakeJacobianPoint(&xVal, &yVal, &zVal)
```

Comment:
shouldn't we check that resulting point is on `secp256k1` curve ? 

I think the easiest way to the parsing here would be to use `btcec.ParseJacobian` that does all those checks

Reply:
Nice, indeed we can just do `ParseJacobian` here